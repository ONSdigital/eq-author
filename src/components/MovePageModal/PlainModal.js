import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Modal from "components/Modal";
import ScrollPane from "components/ScrollPane";

const Fieldset = styled.fieldset`
  border: none;
  padding: 0;
  min-width: 0;
`;

const Title = styled.legend`
  margin: 0;
  padding: 0.75rem;
  color: #858585;
  font-size: 1.25em;
  font-weight: bold;
  display: block;
`;

const BodyWrapper = styled.div`
  max-height: 18em;
`;

const Body = ({ children }) => (
  <ScrollPane>
    <BodyWrapper>{children}</BodyWrapper>
  </ScrollPane>
);

Body.propTypes = {
  children: PropTypes.node.isRequired
};

const PlainModal = styled(Modal).attrs({ hasCloseButton: false })`
  .Overlay {
    background: transparent;
  }

  .Modal {
    width: 23em;
    padding: 0;
    top: 8em;
  }
`;

export default PlainModal;
export { Fieldset, Title, Body };
