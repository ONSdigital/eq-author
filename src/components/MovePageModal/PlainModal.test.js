import React from "react";
import PlainModal, { Body, Fieldset, Title } from "./PlainModal";
import { shallow } from "enzyme";

describe("MovePageModal/PlainModal", () => {
  it("should render", () => {
    const wrapper = shallow(
      <PlainModal isOpen onClose={jest.fn()}>
        <Body>
          <Fieldset>
            <Title>Test</Title>
            <p>
              Lorem ipsum dolor sit, amet consectetur adipisicing elit.
              Reiciendis dignissimos, ipsum laboriosam autem ipsa corrupti dicta
              tempore provident magnam saepe consectetur molestiae illo ad
              temporibus, quasi animi rem aliquid omnis.
            </p>
          </Fieldset>
        </Body>
      </PlainModal>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
