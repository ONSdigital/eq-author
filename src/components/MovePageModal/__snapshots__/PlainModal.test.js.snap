// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovePageModal/PlainModal should render 1`] = `
.c0 .Overlay {
  background: transparent;
}

.c0 .Modal {
  width: 23em;
  padding: 0;
  top: 8em;
}

<Modal
  className="c0"
  hasCloseButton={false}
  isOpen={true}
  onClose={[Function]}
>
  <Body>
    <PlainModal__Fieldset>
      <PlainModal__Title>
        Test
      </PlainModal__Title>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Reiciendis dignissimos, ipsum laboriosam autem ipsa corrupti dicta tempore provident magnam saepe consectetur molestiae illo ad temporibus, quasi animi rem aliquid omnis.
      </p>
    </PlainModal__Fieldset>
  </Body>
</Modal>
`;
